src/services/server/ServerService.ts

```diff
@@ -39,6 +43,26 @@ export class ServerService {
     });
   }
 
+  getServer(server: IServerAddress): ng.IPromise<IServer> {
+    let deferred: ng.IDeferred<IServer> = this.$q.defer<IServer>();
+    this.dmrService.readAttribute({
+      address: this.generateAddress(server),
+      name: "server-state"
+    }).then(serverState =>{
+      if (serverState === "STOPPED") {
+        deferred.resolve(new Server(server, serverState, ''));
```

Comment:
Well, I am not sure @ryanemerson. Server can be STOPPED and we might still want to read information about it. Why not? I would leave this as is and change it to rejecting promise if we really need to.